src/ImageSharp/Formats/Jpeg/GolangPort/Components/Decoder/OrigHuffmanTree.cs

```diff
@@ -246,19 +221,30 @@ public void Dispose()
         /// <returns>The value</returns>
         public int GetValue(int code, int codeLength)
         {
-            return this.Values[this.Indices[codeLength] + code - this.MinCodes[codeLength]];
+            fixed (int* valuesPtr = this.Values)
+            fixed (int* minCodesPtr = this.MinCodes)
+            fixed (int* indicesPtr = this.Indices)
+            {
+                return valuesPtr[indicesPtr[codeLength] + code - minCodesPtr[codeLength]];
+            }
         }
 
-        /// <summary>
-        /// Initializes the Huffman tree
-        /// </summary>
-        private void Init()
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public int ReadLut(int index)
         {
-            this.Lut = IntPool256.Rent(MaxNCodes);
-            this.Values = IntPool256.Rent(MaxNCodes);
-            this.MinCodes = CodesPool16.Rent(MaxCodeLength);
-            this.MaxCodes = CodesPool16.Rent(MaxCodeLength);
-            this.Indices = CodesPool16.Rent(MaxCodeLength);
+            fixed (int* lutPtr = this.Lut)
+            {
+                return lutPtr[index];
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public int GetMaxCode(int index)
+        {
+            fixed (int* maxCodesPtr = this.MaxCodes)
```

Comment:
`GetMaxCode()`  gonna pin/unpin `this.MaxCodes` several times inside a loop. We need to find a better way. Pinning outside the loop, or applying [`ref` arithmetics with `SRCS.Unsafe`](https://github.com/SixLabors/ImageSharp/blob/master/src/ImageSharp/Formats/Jpeg/Common/Block8x8.cs#L52) could do the job.

Do you want me to deal with this?